### PR TITLE
DCCsi Refactor to Use Action Manager

### PR DESCRIPTION
**Pull request by HogJonny-AMZN**
## What does this PR do?

Fixes the DCCsi 'StudioTools' be refactoring menu framework to Action Manager.
https://github.com/o3de/o3de/issues/15180
1. The menu framework is refactored and functions.

Uses these PR's as the basis for the refactor:
https://github.com/o3de/o3de/pull/15239
https://github.com/o3de/o3de/pull/15266

## How was this PR tested?
1. Enabled the Gem in a project, cmake configured and built
2. Started project Editor from Project Manager
3. The 'Studio Tools' menu appears
4. Tested each menu action to ensure previous functionality
5. All apps work (if they are the default supported versions and install locations.)
6. You can now properly bump the Blender version number via settings.local.json without other changes
7. option 1, copy and rename and set the version "DccScriptingInterface\Tools\DCC\Blender\settings.local.json.example"
8. option 2, from cli run:
9. DccScriptingInterface>`.\python.cmd foundation.py` (this makes sure the DCCsi package dependancies are installed and available externally from the Editor, such as when running from the cli.)
10. DccScriptingInterface>`.\python.cmd Tools\DCC\Blender\config.py` and this will generate a full settings.local.json (all settings, but then you have to make sure all paths are updated to match version.)

Editor_tVTQNsNBEP

Editor_ZlrvyJLK7Y

Editor_1nCSuzTWwa

Editor_IABmu40cjE

Is there anything left?  YES I think so ...
1. I need to make sure the the Maya, Blender and Wing start service functions are properly loading their settings.local.json and using that configuration and settings data before launching the applications.

### CODE REVIEW

**Comment by HogJonny-AMZN** [comment]
This PR replaces: https://github.com/o3de/o3de/pull/15394

File: Gems/AtomLyIntegration/TechnicalArt/DccScriptingInterface/CMakeLists.txt [hunk 1]
```diff
@@ -1,9 +1,15 @@
 # 
+
```
Comment (cgalvan) [inline Gems/AtomLyIntegration/TechnicalArt/DccScriptingInterface/CMakeLists.txt:2]: Looks like some extra newlines got introduced in this file

**Review by cgalvan — APPROVED** [review]

**Review by AMZN-daimini — APPROVED** [review]

File: Gems/AtomLyIntegration/TechnicalArt/DccScriptingInterface/Editor/Scripts/bootstrap.py [hunk 2]
```diff
@@ -74,13 +74,13 @@
...
-#DCCSI_GDEBUG = True
-#DCCSI_DEV_MODE = True
+# temporary force enable these during development, as bootstrap is a entrypoint
+DCCSI_GDEBUG = True
+DCCSI_DEV_MODE = True
 
 # enable this if you are having difficulty with debugging
 # the subprocess booting start.py (note: will block editor)
-DCCSI_LOCAL_DEBUG = False # <-- for code branch in this module only
+DCCSI_LOCAL_DEBUG = True # <-- for code branch in this module only
```
[the remark this answers is not in the document]
Comment (HogJonny-AMZN) [reply Gems/AtomLyIntegration/TechnicalArt/DccScriptingInterface/Editor/Scripts/bootstrap.py:83]: nope, tnx

[the remark this answers is not in the document]
Comment (HogJonny-AMZN) [reply Gems/AtomLyIntegration/TechnicalArt/DccScriptingInterface/Editor/Scripts/bootstrap.py]: fixed

**Pull request merged**